drivers/wifi/infineon/airoc_whd_hal_common.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2023 Cypress Semiconductor Corporation (an Infineon company) or
+ * an affiliate of Cypress Semiconductor Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "airoc_whd_hal_common.h"
+#include "airoc_wifi.h"
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+
+LOG_MODULE_DECLARE(infineon_airoc_wifi, CONFIG_WIFI_LOG_LEVEL);
+
+#ifdef __cplusplus
+extern "C" {
```

Comment:
nit (not a blocker): Is this required?

Reply:
Probably not.  In fact, a purist might say that it's wrong to have that in a C source file;  it only belongs in header files.  However, one of my goals was minimum disruption of the original sources ... though that seems to have fallen by the wayside.  The sources airoc_whd_hal_common.c, airoc_whd_hal_sdio.c, and airoc_whd_hal_spi.c were derived from the original airoc_whd_hal.c, and I didn't think about the presence of that #ifdef.  While I'm pretty sure that's superfluous, I'm reluctant to remove it at this late date.  Only if someone insists.

Reply:
This rationale makes sense to me., but I can't resolve the conversation (I would if I could).